src/openms/source/ANALYSIS/QUANTITATION/AbsoluteQuantitationMethod.cpp

```diff
@@ -0,0 +1,259 @@
+  // {
+  //   feature_name_ = feature_name;
+  // }
+  
+  // void AbsoluteQuantitationMethod::getFeatureName(String& feature_name)
+  // {
+  //   feature_name = feature_name_;
+  // } 
+  
+  // void AbsoluteQuantitationMethod::setISName(const String& IS_name)
+  // {
+  //   IS_name_ = IS_name;
+  // }
+  
+  // void AbsoluteQuantitationMethod::getISName(String& IS_name)
+  // {
+  //   IS_name = IS_name_;
+  // }
+  
+  void AbsoluteQuantitationMethod::setConcentrationUnits(const String& concentration_units)
+  {
+    concentration_units_ = concentration_units;
+  }
+
+  void AbsoluteQuantitationMethod::getConcentrationUnits(String& concentration_units)
+  {
+    concentration_units = concentration_units_;
+  }
+  
+  void AbsoluteQuantitationMethod::setTransformationModel(const String& transformation_model, const Param& transformation_model_params)
+  {
+    transformation_model_ = transformation_model;
+    transformation_model_params_ = transformation_model_params;
+  }
+
+  void AbsoluteQuantitationMethod::getTransformationModel(String& transformation_model, Param& transformation_model_params)
+  {
+    transformation_model = transformation_model_;
+    transformation_model_params = transformation_model_params_;
+  }
+  
+  void AbsoluteQuantitationMethod::setActualConcentration(const double& actual_concentration)
+  {
+    actual_concentration_ = actual_concentration;
+  }
+
+  void AbsoluteQuantitationMethod::getActualConcentration(double& actual_concentration)
+  {
+    actual_concentration = actual_concentration_;
+  }
+  
+  void AbsoluteQuantitationMethod::setStatistics(const int& n_points, const double& correlation_coefficient)
+  {
+    n_points_ = n_points;
+    correlation_coefficient_ = correlation_coefficient;
+  }
+  
+  void AbsoluteQuantitationMethod::getStatistics(int& n_points, double& correlation_coefficient)
+  {
+    n_points = n_points_;
+    correlation_coefficient = correlation_coefficient_;
+  }
+
+  Param AbsoluteQuantitationMethod::fitTransformationModel(const String & transformation_model,
+    const TransformationModel::DataPoints& data,
+    const Param& transformation_model_params)
+  {
+    Param params;
+    if (transformation_model == "TransformationModelLinear")
+    {
+      TransformationModelLinear tm(data, transformation_model_params);
+      params = tm.getParameters();
+    }
+    else if (transformation_model == "TransformationModelBSpline")
+    {
+      TransformationModelBSpline tm(data, transformation_model_params);
+      params = tm.getParameters();
+    }
+    else if (transformation_model == "TransformationModelInterpolated")
+    {
+      TransformationModelInterpolated tm(data, transformation_model_params);
+      params = tm.getParameters();
+    }
+    else if (transformation_model == "TransformationModelLowess")
+    {
+      TransformationModelLowess tm(data, transformation_model_params);
+      params = tm.getParameters();
+    }
```

Comment:
@timosachsenberg dont we have a factory fxn to avoid this kind of code? it seems like there should be one ...

Reply:
yep could be solved using a or Factory<> but also adds some boilerplate code.